ios/Exponent/Versioned/Core/Api/EXNotifications.m

```diff
@@ -92,41 +93,124 @@ - (instancetype)initWithExperienceId:(NSString *)experienceId kernelServiceDeleg
+                  rejecter:(__unused RCTPromiseRejectBlock)reject)
+{
+  NSMutableArray<UNNotificationAction *> * actionsArray = [[NSMutableArray alloc] init];
+  
+  for(NSArray *action in actions) {
+    int optionsInt = [(NSNumber *)action[2] intValue];
+    UNNotificationActionOptions options = UNNotificationActionOptionForeground + optionsInt;
+                                     
+    if (([action count] == 5)) {
+      UNTextInputNotificationAction * newAction = [UNTextInputNotificationAction actionWithIdentifier:action[0]
+                                                                                                title:action[1]
+                                                                                              options:options
+                                                                                 textInputButtonTitle:action[3] textInputPlaceholder:action[4]];
+      [actionsArray addObject:newAction];
+    } else {
+      UNNotificationAction * newAction = [UNNotificationAction actionWithIdentifier:action[0] title:action[1] options:options];
+      [actionsArray addObject:newAction];
+    }
+  }
+  
+  UNNotificationCategory * newCategory = [UNNotificationCategory categoryWithIdentifier:categoryId actions:actionsArray intentIdentifiers:@[] options:UNNotificationCategoryOptionNone];
+  
+  [[UNUserNotificationCenter currentNotificationCenter] getNotificationCategoriesWithCompletionHandler:^(NSSet * categories) {
+    NSMutableSet * categoriesMutable = [categories mutableCopy];
+    for( UNNotificationCategory * category in categoriesMutable) {
+      if ([category.identifier isEqualToString:categoryId]) {
+        [categoriesMutable removeObject:category];
+        break;
+      }
+    }
+    [categoriesMutable addObject:newCategory];
+    [[UNUserNotificationCenter currentNotificationCenter] setNotificationCategories:categoriesMutable];
+    resolve(@(YES));
+  }];
 }
 
 RCT_EXPORT_METHOD(scheduleLocalNotification:(NSDictionary *)payload
                   withOptions:(NSDictionary *)options
                   resolver:(RCTPromiseResolveBlock)resolve
                   rejecter:(__unused RCTPromiseRejectBlock)reject)
 {
-  UILocalNotification *notification = [self _localNotificationFromPayload:payload];
-
-  notification.fireDate = [RCTConvert NSDate:options[@"time"]] ?: [NSDate new];
-  notification.repeatInterval = [RCTConvert NSCalendarUnit:options[@"repeat"]] ?: 0;
-
+  bool repeats = NO;
+  if (options[@"repeats"]) {
+    repeats = ([options[@"repeats"] isEqualToString:@"Yes"])? YES : NO;
+  }
+  UNMutableNotificationContent* content = [self _localNotificationFromPayload:payload];
+  
+  NSDateComponents * date = [[NSDateComponents alloc] init];
+  NSArray * unites = @[@"day", @"month", @"year", @"weekday", @"quarter", @"leapMonth", @"nanosecond", @"era", @"weekdayOrdinal", @"weekOfMonth", @"weekOfYear", @"hour", @"second", @"minute", @"yearForWeekOfYear"];
```

Comment:
unites -> units

Some of these seem not so useful or difficult to document. What is "era" or "leapMonth"? We need to support this on Android too.